chain-signatures/contract/src/lib.rs

```diff
@@ -94,29 +111,44 @@ impl SignatureRequest {
 #[derive(BorshDeserialize, BorshSerialize, Debug)]
 pub struct MpcContract {
     protocol_state: ProtocolContractState,
-    pending_requests: LookupMap<SignatureRequest, Option<SignatureResponse>>,
+    pending_requests: LookupMap<SignatureRequest, Option<CryptoHash>>,
     request_counter: u32,
+    yield_resume_requests: LookupMap<u64, YieldResumeRequest>,
```

Comment:
we need them both because we have the need to get the data_id by SignRequest, e.g. getting the data_id from pending_requests, and we also have the need to get SignRequest by yield_resume_index, e.g. in fn sign_on_finish(), we call fn remove_yield_resume_requests(yield_resume_index: u64), which removes both the yield resume request and the SignRequest from pending requests.